cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/BrokerSetAwareGoal.java

```diff
@@ -79,7 +80,7 @@
 public class BrokerSetAwareGoal extends AbstractGoal {
   private static final Logger LOG = LoggerFactory.getLogger(BrokerSetAwareGoal.class);
   private BrokerSetResolutionHelper _brokerSetResolutionHelper;
```

Comment:
I see that the `_brokerSetResolutionHelper` is initialized and stored as a field. What is the lifetime of the BrokerSetAwareGoal class? Is it created each time the goal calculation is done? Or is the same class re-used?

If it is the latter, then having a `BrokerSetResolutionHelper` as a member will have the side effect of re-using stale broker to color mappings even as they change.

Reply:
Those goals are indeed initialized multiple times from different places, for different purposes. We don't need to worry about that as in our case, the broker assignment is not called during goal init time.

However, you're right that the goal will re-calculate brokerSet assignment on each goal optimization. We think that is the expected behavior. The user of this goal should take care of the broker set cache.